contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1727,6 +1730,204 @@ object_name(PG_FUNCTION_ARGS)
 	PG_RETURN_NULL();
 }
 
+/*
+ * type_id
+ * Returns the object ID with type name as input.
+ * Returns NULL
+ *  if input is NULL
+ *  if there is no such type
+ *  if user don't have right permission
+ *  if any error occured
+ */
+Datum
+type_id(PG_FUNCTION_ARGS)
+{
+    char       *db_name,
+               *schema_name,
+               *object_name;
+    char       *physical_schema_name;
+    char       *input;
+    char       **splitted_object_name;
+    Oid        schema_oid;
+    Oid        user_id = GetUserId();
+    Oid        result = InvalidOid;
+    int        i;
+
+    if (PG_ARGISNULL(0))
+        PG_RETURN_NULL();
+    input = text_to_cstring(PG_GETARG_TEXT_PP(0));
+
+    /* strip trailing whitespace from input */
+    i = pg_mbstrlen(input);
+    while (i > 0 && isspace((unsigned char) input[i - 1]))
```

Comment:
Can't there be spaces before the starting of the input? I mean can't we call this function like ```SELECT TYPE_ID('   type_a  ');```
If so, we need update this condition as well.

Reply:
Spaces in the starting is not supported as per tsql behaviour.